modules/basics/src/main/java/com/opengamma/strata/basics/currency/Money.java

```diff
@@ -0,0 +1,223 @@
+ * it should be assumed that the numbers are an approximation, and not an exact figure.
+ * <p>
+ * This class is immutable and thread-safe.
+ */
+public class Money implements FxConvertible<Money>, Comparable<Money>, Serializable {
+
+  /** Serialization version. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * The currency.
+   * For example, in the value 'GBP 12.34' the currency is 'GBP'.
+   */
+  private final Currency currency;
+
+  /**
+   * The amount of the currency.
+   * For example, in the value 'GBP 12.34' the amount is 12.34.
+   */
+  private final BigDecimal amount;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance of {@code Money} for the specified currency and amount.
+   *
+   * @param currencyAmount the instance of {@link CurrencyAmount} wrapping the currency and amount.
+   * @return the currency amount
+   */
+  public static Money of(CurrencyAmount currencyAmount) {
+    Currency currency = currencyAmount.getCurrency();
+    BigDecimal roundedAmount = new BigDecimal(currencyAmount.getAmount())
+        .setScale(currency.getMinorUnitDigits(), BigDecimal.ROUND_HALF_EVEN);
+    return new Money(currency, roundedAmount);
+  }
+
+  /**
+   * Obtains an instance of {@code Money} for the specified currency and amount.
+   *
+   * @param currency the currency the amount is in
+   * @param amount the amount of the currency to represent
+   * @return the currency amount
+   */
+  public static Money of(Currency currency, double amount) {
+    BigDecimal roundedAmount = new BigDecimal(amount)
+        .setScale(currency.getMinorUnitDigits(), BigDecimal.ROUND_HALF_EVEN);
+    return new Money(currency, roundedAmount);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates an instance.
+   *
+   * @param currency the currency
+   * @param amount the amount
+   */
+  private Money(Currency currency, BigDecimal amount) {
+    this.currency = currency;
+    this.amount = amount;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the currency.
+   * <p>
+   * For example, in the value 'GBP 12.34' the currency is 'GBP'.
+   *
+   * @return the currency
+   */
+  public Currency getCurrency() {
+    return currency;
+  }
+
+  /**
+   * Gets the amount of the currency as an instance of {@link BigDecimal}.
+   * The amount will be rounded to the currency specifications.
+   * <p>
+   * For example, in the value 'GBP 12.34' the amount is 12.34.
+   *
+   * @return the amount
+   */
+  public BigDecimal getAmount() {
+    return amount;
+  }
+
+  /**
+   * Gets the amount of the currency as a double primitive.
+   * The amount will be rounded to the currency specifications.
+   * <p>
+   * For example, in the value 'GBP 12.34' the amount is 12.34.
+   *
+   * @return the amount
+   */
+  public double getAmountAsDouble() {
```

Comment:
Not sure this method is worth it. Just call

    money.getAmount().doubleValue();